### PR TITLE
Simplify NotNullIfParameterNotNull implementation in PEParameterSymbol

### PR DESCRIPTION
**Pull request by Youssef1313**
See https://github.com/dotnet/roslyn/pull/51952#discussion_r597893509

### CODE REVIEW

File: src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEParameterSymbol.cs
Comment (333fred) [inline src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEParameterSymbol.cs:709]: We should instead do what we do for the other nullable attributes: http://sourceroslyn.io/#Microsoft.CodeAnalysis/MetadataReader/PEModule.cs,1186

Comment (Youssef1313) [reply src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEParameterSymbol.cs]: Not sure if the last commit is what you meant.

File: src/Compilers/Core/Portable/MetadataReader/PEModule.cs
Comment (333fred) [inline src/Compilers/Core/Portable/MetadataReader/PEModule.cs:1204]: You can use the `FindTargetAttribute` version from this type.

File: src/Compilers/Core/Portable/MetadataReader/PEModule.cs
Comment (333fred) [inline src/Compilers/Core/Portable/MetadataReader/PEModule.cs:1216]: You won't need to loop through with the above change, but in the future, we'd generally break at this point.

File: src/Compilers/Core/Portable/MetadataReader/PEModule.cs
Comment (Youssef1313) [reply src/Compilers/Core/Portable/MetadataReader/PEModule.cs:1204]: @333fred Don't we want all the attributes?

Comment (333fred) [reply src/Compilers/Core/Portable/MetadataReader/PEModule.cs:1204]: Oh, interesting, I didn't realize that `NotNullIfNotNull` had `AllowMultiple = true`. In that case, never mind.

**Review by 333fred — APPROVED** [review]
LGTM (commit 4), assuming tests pass. @dotnet/roslyn-compiler for a second review please.

**Review by RikkiGibson — APPROVED** [review]

**Review by cston — APPROVED** [review]

**Comment by 333fred** [comment]
Thanks @Youssef1313!

**Pull request merged**
